tirocks/src/db/imp.rs

```diff
@@ -71,6 +73,41 @@ where
     }
 }
 
+struct PostWriteCallback<'a, F: FnMut()> {
+    raw: SimplePostWriteCallback,
+    _callback: &'a mut F,
+}
+
+extern "C" fn on_post_write_callback<F: FnMut()>(ctx: *mut c_void) {
+    unsafe {
+        let ctx = &mut *(ctx as *mut F);
+        ctx();
+    }
+}
+
+impl<'a, F: FnMut()> PostWriteCallback<'a, F> {
+    #[inline]
+    fn new(callback: &'a mut F) -> Self {
```

Comment:
Why not move it?

Reply:
My initial thought is to reuse custom callback that implements `Fn`. But judging from the bench results I guess native closure is good enough.

Reply:
For this specific spot, I think using reference is better. Because in theory `crocksdb_simple_post_write_callback_init` cannot reliably reference the callback stored in a structure that is about to be move returned.